Source/Engine/Content/Factories/IAssetFactory.h

```diff
@@ -67,3 +67,32 @@ class FLAXENGINE_API IAssetFactory
         return nullptr;
     }
 };
+
+#if NDEBUG
+#define INTERNAL_FACTORY_REGISTER_CHEAK(type)\
+if (IAssetFactory::Get().ContainsKey(type::TypeName)) \
```

Comment:
`CHECK(!IAssetFactory::Get().ContainsKey(type::TypeName))`

Reply:
custom massage better suits this case 
giving better information was is wrong

Reply:
Then at least I would skip using `#type` to reduce unique strings in compiled binary (smaller engine size).